pkg/controller/disruption/disruption.go

```diff
@@ -0,0 +1,587 @@
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Policy().PodDisruptionBudgets(api.NamespaceAll).Watch(options)
+			},
+		},
+		&policy.PodDisruptionBudget{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    dc.addDb,
+			UpdateFunc: dc.updateDb,
+			DeleteFunc: dc.removeDb,
+		},
+	)
+	dc.pdbLister.Store = dc.pdbStore
+
+	dc.rcIndexer, dc.rcController = framework.NewIndexerInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.ReplicationControllers(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.ReplicationControllers(api.NamespaceAll).Watch(options)
+			},
+		},
+		&api.ReplicationController{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+
+	dc.rcLister.Indexer = dc.rcIndexer
+
+	dc.rsStore, dc.rsController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.Extensions().ReplicaSets(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Extensions().ReplicaSets(api.NamespaceAll).Watch(options)
+			},
+		},
+		&extensions.ReplicaSet{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+	)
+
+	dc.rsLister.Store = dc.rsStore
+
+	dc.dStore, dc.dController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return dc.kubeClient.Extensions().Deployments(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return dc.kubeClient.Extensions().Deployments(api.NamespaceAll).Watch(options)
+			},
+		},
+		&extensions.Deployment{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+	)
+
+	dc.dLister.Store = dc.dStore
+
+	return dc
+}
+
+// TODO(mml): When controllerRef is implemented (#2210), we *could* simply
+// return controllers without their scales, and access scale type-generically
+// via the scale subresource.  That may not be as much of a win as it sounds,
+// however.  We are accessing everything through the pkg/client/cache API that
+// we have to set up and tune to the types we know we'll be accessing anyway,
+// and we may well need further tweaks just to be able to access scale
+// subresources.
+func (dc *DisruptionController) finders() []podControllerFinder {
+	return []podControllerFinder{dc.getPodReplicationControllers, dc.getPodDeployments, dc.getPodReplicaSets}
+}
+
+/* getPodReplicaSets finds replicasets which have no matching deployments.
+ */
+func (dc *DisruptionController) getPodReplicaSets(pod *api.Pod) ([]controllerAndScale, error) {
```

Comment:
Well, I *am* returning the ReplicaSet, but I'm also returning the scale in such a way that the caller doesn't need to know how to deal with a ReplicaSet, or a ReplicationController.

Also there may be more than one controller, and that's a condition we need to be aware of.

I could have returned a map instead of a list of pairs, but this seemed natural and perfectly fine.